client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -73,7 +81,7 @@ export default React.createClass( {
 		const sanitizedFieldValues = Object.assign( {}, fieldValues );
 		this.fieldNames.forEach( ( fieldName ) => {
 			if ( typeof fieldValues[ fieldName ] === 'string' ) {
-				sanitizedFieldValues[ fieldName ] = fieldValues[ fieldName ].trim();
+				sanitizedFieldValues[ fieldName ] = deburr( fieldValues[ fieldName ].trim() );
```

Comment:
I'd rather not - the `sanitizerFunction` feature of `formState.Controller` is there for a reason, I'd prefer not to do any sanitizing somewhere else. But maybe that functionality can be improved in `form-state` itself - though, after having a quick glance there, I'm not sure why there's a delay before the sanitized values are shown.

> BTW: Is it really required to type in latin letters? That seems a huge oversight for i18n..
It's domains, what else did you expect? 😀 The [official RFC for WHOIS](https://tools.ietf.org/html/rfc3912) admits that it's lacking explicit i18n support:

> 4.  Internationalisation
> 
>    The WHOIS protocol has not been internationalised.  The WHOIS
>    protocol has no mechanism for indicating the character set in use.
>    Originally, the predominant text encoding in use was US-ASCII.  In
>    practice, some WHOIS servers, particularly those outside the USA,
>    might be using some other character set either for requests, replies,
>    or both.  This inability to predict or express text encoding has
>    adversely impacted the interoperability (and, therefore, usefulness)
>    of the WHOIS protocol.

I've checked and, for example, the Polish registry for `.pl` indeed allows Polish characters to be used in the WHOIS. But with our top TLDs handled by Verisign, I'm guessing that's not the case.